framework/include/base/FEProblem.h

```diff
@@ -481,12 +481,12 @@ class FEProblem :
    * @return Const reference to the user object
    */
   template <class T>
-  const T & getUserObject(const std::string & name)
+  const T & getUserObject(const std::string & name, unsigned int tid = 0)
```

Comment:
Due to the discussion in https://github.com/libMesh/libmesh/issues/801, this PR is DOA. We cannot retrieve thread specific versions of an object and expect them to contain specific data.

Tagging @friedmud